src/cdk-experimental/table/virtual-rows.ts

```diff
@@ -0,0 +1,292 @@
+  VIRTUAL_SCROLL_STRATEGY,
+  VirtualScrollStrategy
+} from '@angular/cdk/scrolling';
+
+
+/**
+ * Union of the types that can be set as the data source for a `CdkTable`.
+ * @docs-private
+ */
+type CdkTableDataSourceInput<T> =
+    DataSource<T>|Observable<ReadonlyArray<T>|T[]>|ReadonlyArray<T>|T[];
+
+
+class PositioningListenerProxy implements StickyPositioningListener {
+  private _listener?: StickyPositioningListener;
+
+  setListener(listener: StickyPositioningListener) {
+    this._listener = listener;
+  }
+
+  stickyColumnsUpdated(update: StickyUpdate): void {
+    this._listener?.stickyColumnsUpdated(update);
+  }
+
+  stickyEndColumnsUpdated(update: StickyUpdate): void {
+    this._listener?.stickyEndColumnsUpdated(update);
+  }
+
+  stickyFooterRowsUpdated(update: StickyUpdate): void {
+    this._listener?.stickyFooterRowsUpdated(update);
+  }
+
+  stickyHeaderRowsUpdated(update: StickyUpdate): void {
+    this._listener?.stickyHeaderRowsUpdated(update);
+  }
+}
+
+
+/**
+ * A directive that enables virtual scroll for a {@link CdkTable}.
+ */
+@Directive({
+  selector: 'cdk-table[virtualRows], table[cdk-table][virtualRows]',
+  exportAs: 'cdkVirtualRows',
+  providers: [
+    {provide: _VIEW_REPEATER_STRATEGY, useClass: _RecycleViewRepeaterStrategy},
+    // The directive cannot provide itself as the sticky positions listener because it introduces
+    // a circular dependency. Use an intermediate listener as a proxy.
+    {provide: STICKY_POSITIONING_LISTENER, useClass: PositioningListenerProxy},
+  ],
+})
+export class CdkVirtualRows<T> implements CdkVirtualScrollRepeater<T>, OnDestroy, StickyPositioningListener {
+  /** Emits when the component is destroyed. **/
+  private _destroyed = new ReplaySubject<void>(1);
+
+  /** The currently rendered range of indices. */
+  private _renderedRange: ListRange;
+
+  /** Subject that emits when a new DataSource instance is given. */
+  private _dataSourceChanges = new Subject<CdkTableDataSourceInput<T>>();
+
+  /** The original data source passed from the parent component. */
+  private _innerDataSource: CdkTableDataSourceInput<T>;
+
+  /** Emits when the virtual scroll viewport's CSS `translateY()` value changes. */
+  private _renderedContentOffsetChanged = this._viewport.renderedContentOffsetChanged.pipe(
+      map(() => this._viewport.getOffsetToRenderedContentStart()),
+      filter(offset => offset !== null) as OperatorFunction<number|null, number>,
+      distinctUntilChanged());
+
+  /** Emits when the header rows sticky state changes. */
+  private _headerRowStickyUpdates = new Subject<StickyUpdate>();
+
+  /** Emits when the footer rows sticky state changes. */
+  private _footerRowStickyUpdates = new Subject<StickyUpdate>();
+
+  /**
+   * Observable that emits the data source's complete data set. This exists to implement
+   * {@link CdkVirtualScrollRepeater}.
+   */
+  readonly dataStream: Observable<T[] | ReadonlyArray<T>> =
+      this._dataSourceChanges.asObservable()
```

Comment:
You don't need the `asObservable` call, the `pipe` already turns it into an observable.

Reply:
Done